### ISSUE TITLE
Failed to register jupyter_bokeh

### ISSUE THREAD

**Issue by obrown06** [bokeh/jupyter_bokeh#201]
I'm trying to integrate a `Tabulator` into a Voila application as follows:

```
import pandas as pd
import panel as pn
pn.extension('tabulator')

df = pd.DataFrame(...)
df_widget = pn.ipywidget(pn.widgets.Tabulator(df))
```

I'm getting the following rendered error:

```
Failed to load model class 'BokehModel' from module '@bokeh/jupyter_bokeh'
```

And the following errors in the javascript console:

```
Plugin '@bokeh/jupyter_bokeh' failed to activate.
TypeError: No provider for: jupyter.extensions.jupyterWidgetRegistry.
Error: No version of module @bokeh/jupyter_bokeh is registered
```

The relevant package versions:

```
bokeh                     3.4.0
ipywidgets                8.1.2
jupyter                   1.0.0
jupyter_bokeh             4.0.1
jupyterlab                4.0.11
jupyterlab_widgets        3.0.10
panel                     1.4.1
python                    3.10.14
```

Anyone have an idea what might be going wrong here? Happy to redirect the query elsewhere if this isn't the right place. Thanks!

**Comment by bryevdv** [comment]
According to a [Jupyter core dev](https://discourse.jupyter.org/t/lab-v3-no-provider-for-jupyter-extensions-jupyterwidgetregistry/7789/2)

> That indicates that the Jupyter widgets extension is not installed. Assuming your server is running jlab 3, it can be installed by installing ipywidgets 7.6+ or by installing jupyterlab_widgets 1.0+.

Is it possible you have an environment mixup and `jupyterlab_widgets` is not installed in the same environment you think it is?

**Comment by obrown06** [comment]
FWIW: I'm using Conda and it seems like `jupyterlab_widgets` is definitely installed:

```
conda list | grep jupyterlab_widgets
jupyterlab_widgets        3.0.10
```

**Comment by obrown06** [comment]
Also, it looks like at least someone else may be hitting the same issue: https://community.deeplearning.ai/t/replicating-chatbot-implementation-with-huggingface-open-source-models/596117.

**Comment by Kalandoros** [comment]
I got the same issue.  This is not new issue.
It's already persising for longer 

jupyter_bokeh                     4.0.4
jupyter_client                    8.1.0
jupyter_core                      5.3.0
jupyter-events                    0.7.0
jupyter-lsp                       2.2.0
jupyter_server                    2.7.3
jupyter_server_terminals          0.4.4
jupyterlab                        4.2.1
jupyterlab-pygments               0.2.2
jupyterlab_server                 2.27.2
jupyterlab_widgets                3.0.10
panel                             1.4.3
voila                             0.5.7

labextensions:
 @bokeh/jupyter_bokeh v4.0.4 enabled ok (python, jupyter_bokeh)
@pyviz/jupyterlab_pyviz v3.0.2 enabled ok

![image](https://github.com/bokeh/jupyter_bokeh/assets/129214458/0031c3b2-99ed-404b-b4a3-0ddaf5b5c7d5)

**Comment by MarcSkovMadsen** [comment]
+1. My colleague has the same issue

**Comment by MarcSkovMadsen** [comment]
I have managed to get this working in one case. What is below

- Made sure I have the latest versions of Python and Jupyter Extension. I had to upgrade Jupyter extension.
- Installed the kernel of my python environment: `python -m ipykernel install --user --name myenv --display-name "Python (myenv)"`.
- Created and run the notebook in jupyterlab inspired by a stackoverflow response
- Opened VS Code (Code Server) in my browser instead of a standalone PWA app. Ran the code. For some reason it here asked me to install [**Jupyter Notebook Renderers**](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter-renderers). I installed the renderer.

After each step I tried. Only after the last step it worked.

I will try to find a receipe that will work from scratch

**Comment by MarcSkovMadsen** [comment]
In another JupyterHub I see the below in my browser console

```bash
On another instance of VS Code I see

Error: Unable to read file 'vscode-remote://MY.DOMAIN/vscode-resource?{"requestResourcePath":"/home/jovyan/.local/share/code-server/extensions/ms-toolsai.jupyter-2024.3.1-universal/temp/scripts/af7cfb460b26abe4dbd921151f82f0407a7a48377d4530dbeb78bbfab664c42b/jupyter/@bokeh/jupyter_bokeh.js"}' (Error: Unable to resolve nonexistent file 'vscode-remote://mnr-jupyterhub.de-prod.dk/vscode-resource?{"requestResourcePath":"/home/jovyan/.local/share/code-server/extensions/ms-toolsai.jupyter-2024.3.1-universal/temp/scripts/af7cfb460b26abe4dbd921151f82f0407a7a48377d4530dbeb78bbfab664c42b/jupyter/@bokeh/jupyter_bokeh.js"}')
    at n.F (fileService.ts:631:10)
    at n.D (fileService.ts:616:15)
    at async v (resourceLoading.ts:68:18)
    at async M.yb (webviewElement.ts:743:19)
```

I don't believe the path `MY.DOMAIN/vscode-resource` is valid. I.e. there is some proxy issue.

**Comment by bryevdv** [comment]
cc @philippjfr for any thoughts

**Comment by MarcSkovMadsen** [comment]
I was not able to use steps above to find solution on another JupyterHub.

**Comment by contang0** [comment]
panel version 1.4.5 does not have this issue. I installed the latest version, and running pn.extension() triggers the following error:

```
Unable to find widget '@bokeh/jupyter_bokeh' version '^4.0.5' from configured widget sources ["jsdelivr.com","unpkg.com"]. Expected behavior may be affected. Click here for more information.
Sources
```

Downgraded to 1.4.5 and the error is gone.

**Comment by philippjfr** [comment]
Sorry, missed this entirely. Will have to do some digging because 4.0.5 definitely does exist on `npm`:

https://www.npmjs.com/package/@bokeh/jupyter_bokeh?activeTab=versions

**Comment by samimia-swks** [comment]
Possibly the same issue : https://github.com/bokeh/jupyter_bokeh/issues/209

**Comment by jwintel** [comment]
I had a similar issue and found the answer [here](https://github.com/bokeh/jupyter_bokeh/issues/209): the link at the bottom of the issue "Click [here](https://github.com/microsoft/vscode-jupyter/wiki/IPyWidget-Support-in-VS-Code-Python) for more information" contained settings that weren't in my vscode.

Once I added "jupyter.widgetScriptSources": ["jsdelivr.com", "unpkg.com"], to my vscode settings widgets started rendering in the interactive vscode for me.
